schema-nft-marketplace.graphql

```diff
@@ -0,0 +1,160 @@
+# Subgraph Schema: NFT Marketplace
+# Version: 0.0.1
+
+enum Network {
+  ARBITRUM_ONE
+  ARWEAVE_MAINNET
+  AURORA
+  AVALANCHE
+  BOBA
+  BSC # aka BNB Chain
+  CELO
+  COSMOS
+  CRONOS
+  MAINNET # Ethereum Mainnet
+  FANTOM
+  FUSE
+  HARMONY
+  JUNO
+  MOONBEAM
+  MOONRIVER
+  NEAR_MAINNET
+  OPTIMISM
+  OSMOSIS
+  MATIC # aka Polygon
+  XDAI # aka Gnosis Chain
+}
+
+enum NftStandard {
+    ERC721
+    ERC721A
+    ERC1155
+}
+
+enum SaleStrategy {
+    " Strategy that executes an order at a fixed price that can be taken either by a bid or an ask. "
+    StandardSale
+
+    " Strategy that executes an order at a fixed price that can be matched by any tokenId for the collection. "
+    AnyItemFromCollection
+
+    " Strategy that executes an order at a fixed price that can be matched by any tokenId in a set of tokenIds. "
+    AnyItemFromSet
+
+    " Strategy to launch a Dutch Auction for a token where the price decreases linearly until a specified timestamp and end price defined by the seller. "
+    DutchAuction
+
+    " Strategy to set up an order that can only be executed by a specific address. "
+    PrivateSale
+}
+
+type Marketplace @entity {
+    " Smart contract address of the protocol's main contract (Factory, Registry, etc) "
+    id: ID!
+
+    " Name of the NFT marketplace, for example LooksRare "
+    name: String!
+
+    " The blockchain network this subgraph is indexing on. "
+    network: Network!
+
+    " Number of collections listed on the marketplace. "
+    collectionCount: Int!
+
+    " Cumulative sale volume (in ETH) "
+    cumulativeSaleVolumeETH: BigDecimal!
+
+    " Cumulative sale volume (in USD)"
+    cumulativeSaleVolumeUSD: BigDecimal!
+
+    " Revenue that goes to the marketplace protocol, aka protocol fee. "
+    marketplaceRevenue: BigDecimal!
+
+    " Revenue that goes to creator, aka royalty fee. "
+    creatorRevenue: BigDecimal!
+
+    " Sum of marketplaceRevenue and creatorRevenue. "
+    totalRevenue: BigDecimal!
+
+    " Trade count of the all collections on the marketplace. "
+    totalTradeCount: Int!
+}
+
+type Collection @entity {
+    " Contract address. "
+    id: ID!
+
+    " Collectio name, mirrored from the smart contract. "
+    name: String!
```

Comment:
Most of the fields here including `name` and `symbol` are not required by the ERC721 interface. Are we going to give them a default value if there isn't one? Or should we leave them as optional?
https://eips.ethereum.org/EIPS/eip-721